File: tests/models/ShoppingCart.ts
import { z } from 'zod';

export const CartItemSchema = z.object({
  productId: z.number(),
  productName: z.string(),
  unitPrice: z.number(),
  quantity: z.number(),
});

export type CartItem = z.infer<typeof CartItemSchema>;

export const ShoppingCartSchema = z.object({
  id: z.number(),
  customerId: z.number(),
  createdAt: z.string(),
  expiresAt: z.string().nullable(),
  items: z.array(CartItemSchema),
});

export type ShoppingCart = z.infer<typeof ShoppingCartSchema>;

File: api/OnlineStore.Api/Models/ShoppingCart.cs
namespace OnlineStore.Api.Models;

public class ShoppingCart
{
    public int Id { get; set; }
    public int CustomerId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public List<CartItem> Items { get; set; } = new();
}

public class CartItem
{
    public int ProductId { get; set; }
    public string ProductName { get; set; } = string.Empty;
    public decimal UnitPrice { get; set; }
    public int Quantity { get; set; }
}
